### PR TITLE
Clean up the SMB/CIFS documentation

### PR DESCRIPTION
**Pull request by settermjd**
As part of making the changes required for #411, the SMB/CIFS documentation needed to be cleaned up. This commit makes the required changes.

### CODE REVIEW

File: modules/administration_manual/pages/configuration/files/external_storage/smb.adoc [hunk 1]
```diff
@@ -1,48 +1,58 @@
...
+It is included in all Linux distributions; on _Debian_, _Ubuntu_, and other Debian
+derivatives this is `smbclient`. 
+On _SUSE_, _Red Hat_, _CentOS_, and other Red Hat derivatives it is `samba-client`. 
+You also need `which` and `stdbuf`, which should be included in most Linux distributions.
+
+== Configuration
+
+When configuring it, you will need the following information:
+
+* The folder name for your local mountpoint.
```
Comment (mmattel) [inline modules/administration_manual/pages/configuration/files/external_storage/smb.adoc]: Proposal:
A name for your local mountpoint which shows up as a folder

File: modules/administration_manual/pages/configuration/files/external_storage/smb.adoc [hunk 2]
```diff
@@ -1,48 +1,58 @@
...
+* The URL of the Samba server.
+* The username or domain/username used to login to the Samba server.
+* the password to login to the Samba server.
```
Comment (mmattel) [inline modules/administration_manual/pages/configuration/files/external_storage/smb.adoc]: Capital `T`

File: modules/administration_manual/pages/configuration/files/external_storage/smb.adoc [hunk 3]
```diff
@@ -1,48 +1,58 @@
...
+* The share name to mount on the remote Samba server.
+* The remote subfolder inside the remote Samba share to mount. This is optional, as it defaults to `/`.
++
+TIP: To assign the ownCloud logon username automatically to the subfolder, use `$user` instead of a subfolder name.
+* The ownCloud users and groups who get access to the share.
+
+[TIP]
+====
+Optionally, you can specify a `Domain`. 
+This is useful in cases where the SMB server requires a domain and a username, and an advanced authentication mechanism like session credentials is used so that the username cannot be modified.
```
Comment (mmattel) [inline modules/administration_manual/pages/configuration/files/external_storage/smb.adoc]: ...and an advanced authentication mechanism like Active Directory (AD) or when using session credentials where the username cannot be modified.

File: modules/administration_manual/pages/configuration/files/external_storage/smb.adoc (hunk 1 above)
Comment (settermjd) [reply modules/administration_manual/pages/configuration/files/external_storage/smb.adoc]: On thinking about it further, "The folder name, which will be your local mount point." will work better.

**Pull request merged**